block/manager.go

```diff
@@ -330,12 +396,73 @@ func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
 		if err != nil {
 			m.logger.Error("failed to save updated state", "error", err)
 		}
-		delete(m.syncCache, currentHeight+1)
+		m.blockCache.deleteBlock(currentHeight + 1)
 	}
 
 	return nil
 }
 
+// BlockStoreRetrieveLoop is responsible for retrieving blocks from the Block Store.
+func (m *Manager) BlockStoreRetrieveLoop(ctx context.Context) {
+	// waitCh is used to signal the block store retrieve loop, that it should check block store for new blocks
+	// blockStoreCond can be signalled in completely async manner, and goroutine below
+	// works as some kind of "buffer" for those signals
+	waitCh := make(chan interface{})
+	lastBlockStoreHeight := uint64(0)
+	go func() {
+		for {
+			// This infinite loop is expected to be stopped once the context is
+			// cancelled or throws an error and cleaned up by the GC. This is OK
+			// because it waits using a conditional which is only signaled periodically.
+			m.blockStoreMtx.Lock()
+			m.blockStoreCond.Wait()
```

Comment:
I think this infinite loop can be avoided by using a channel instead of a `sync.Cond`. As we don't have a condition between `Lock` and `Wait()` we don't need a conditional sync.

https://lukechampine.com/cond.html

Reply:
yes good point, definitely worth exploring. The same thing applies to `RetrieveLoop` which is already existing code separate from this PR. Maybe we can explore doing this in a separate PR? 😃 

Created an issue for this: https://github.com/rollkit/rollkit/issues/1132

Reply:
Also, side note since it does not apply here, we should be mindful of trying to use channels everywhere instead of mutexes.

> A common Go newbie mistake is to over-use channels and goroutines just because it's possible, and/or because it's fun. Don't be afraid to use a [sync.Mutex](https://pkg.go.dev/sync/#Mutex) if that fits your problem best. Go is pragmatic in letting you use the tools that solve your problem best and not forcing you into one style of code.

Source: https://github.com/golang/go/wiki/MutexOrChannel

Reply:
OK.